packages/app/src/systems/Ecosystem/hooks/useEcosystem.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect } from 'react';
+
+import type { EcosystemInputs, EcosystemMachineState } from '../machines';
+
+import { Services, store } from '~/store';
+
+const selectors = {
+  context: (state: EcosystemMachineState) => state.context,
+  filteredProjects: (state: EcosystemMachineState) =>
+    state.context?.filteredProjects,
+  tags: (state: EcosystemMachineState) => state.context?.tags,
+  filter: (state: EcosystemMachineState) => state.context?.filter,
+  search: (state: EcosystemMachineState) => state.context?.search,
+  isLoading: (state: EcosystemMachineState) =>
+    state.matches('fetching') ||
+    state.matches('searching') ||
+    state.matches('filtering'),
+};
```

Comment:
a good strategy is also using `tags` 

```suggestion
  isLoading: (state: EcosystemMachineState) =>
    state.hasTag('isLoading')
};
```

then you can add to your machine:
```
states: {
  one: {
    tags: ['isLoading']
  },
  two: {
    tags: ['isLoading']
  },
}
```